sw/device/silicon_creator/rom/doc/sigverify.md

```diff
@@ -0,0 +1,183 @@
+2. Compute the digest of the image.
+3. Perform modular exponentiation to compute the encoded message.
+4. Check the encoded message.
+
+### Public Keys
+
+Manifests[^2] of in-flash boot stage images, e.g. ROM\_EXT or the first owner boot stage, contain the public key (exponent and modulus) for verifying their signatures.
+While this information is useful for checking the integrity of an image off-target, e.g. using a developer utility, it's trivial for an attacker to sign an image with an arbitrary key.
+In order to establish the authenticity of a manifest, sigverify uses the authorized public keys stored in ROM for signature verification.
+
+Sigverify can currently store up to 8 authorized keys of 3 different types (roles): Test (manufacturing), dev (development), and prod (production).
+Each authorized key has a corresponding byte (`hardened_byte_bool_t`) in the `CREATOR_SW_CFG_KEY_IS_VALID` OTP item that can be used to irreversibly invalidate that key during manufacturing in case the corresponding private key is compromised.
+Validity of these keys are further gated by the life cycle state of the device: Test keys can only be used in TEST\_UNLOCKED and RMA life cycle states, dev keys can only be used in the DEV life cycle state, and prod keys can be used in all life cycle states.
+The public key of a manifest must be both authorized and valid, i.e. not invalidated in OTP and allowed in the current life cycle state of the device, for sigverify to attempt to verify the signature of an image.
+The following table summarizes the conditions under which each type of key is valid.
+Note that key validity during manufacturing (TEST\_UNLOCKED) does not depend on the OTP value since it may not have been programmed yet.
+
+<table>
+  <tr>
+   <td style="background-color: null">
+   </td>
+   <td style="background-color: #f3f3f3">TEST_UNLOCKED
+   </td>
+   <td style="background-color: #f3f3f3">PROD, PROD_END
+   </td>
+   <td style="background-color: #f3f3f3">DEV
+   </td>
+   <td style="background-color: #f3f3f3">RMA
+   </td>
+  </tr>
+  <tr>
+   <td style="background-color: #f3f3f3">Test keys
+   </td>
+   <td style="background-color: null">Yes
+   </td>
+   <td style="background-color: null">No
+   </td>
+   <td style="background-color: null">No
+   </td>
+   <td style="background-color: null">OTP
+   </td>
+  </tr>
+  <tr>
+   <td style="background-color: #f3f3f3">Dev keys
+   </td>
+   <td style="background-color: null">No
+   </td>
+   <td style="background-color: null">No
+   </td>
+   <td style="background-color: null">OTP
+   </td>
+   <td style="background-color: null">No
+   </td>
+  </tr>
+  <tr>
+   <td style="background-color: #f3f3f3">Prod keys
+   </td>
+   <td style="background-color: null">Yes
+   </td>
+   <td style="background-color: null">OTP
+   </td>
+   <td style="background-color: null">OTP
+   </td>
+   <td style="background-color: null">OTP
+   </td>
+  </tr>
+</table>
```

Comment:
This could be a markdown table which makes it readable from the text file, though you can't make columns into headers. It would look like this:

![image](https://user-images.githubusercontent.com/105280833/231780376-1ae56fa2-4760-4cde-bf89-811168096869.png)

```suggestion
|               | TEST\_UNLOCKED | PROD, PROD_END | DEV | RMA |
|--------------:|----------------|----------------|-----|-----|
| **Test keys** | Yes            | No             | No  | OTP |
|  **Dev keys** | No             | No             | OTP | No  |
| **Prod keys** | Yes            | OTP            | OTP | OTP |

```